modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/model/InfrastructureElementModelDelegate.java

```diff
@@ -155,21 +159,48 @@ public void visitSimpleValue(ParameterSimpleValue text) {
 
               @Override
               public void visitObjectValue(ParameterObjectValue objectValue) {
-                if (!(TLS_KEY_STORE_ELEMENT_IDENTIFIER.equals(name) || TLS_TRUST_STORE_ELEMENT_IDENTIFIER.equals(name))) {
+                if (!(TLS_KEY_STORE_ELEMENT_IDENTIFIER.equals(name)
+                    || TLS_TRUST_STORE_ELEMENT_IDENTIFIER.equals(name)
+                    || TLS_REVOCATION_CHECK_ELEMENT_IDENTIFIER.equals(name))) {
+
                   if (LOGGER.isDebugEnabled()) {
                     LOGGER.debug(format("Skipping unknown parameter with name [%s] for TLSContext", name));
                   }
                   return;
                 }
 
-                InternalComponentConfiguration.Builder nested = InternalComponentConfiguration.builder()
-                    .withIdentifier(builder()
-                        .namespace(TLS_PREFIX)
-                        .name(name)
-                        .build());
 
-                cloneParameters(objectValue, nested);
-                tlsConfig.withNestedComponent(nested.build());
+                if (TLS_REVOCATION_CHECK_ELEMENT_IDENTIFIER.equals(name)) {
+                  InternalComponentConfiguration.Builder revocationCheck = InternalComponentConfiguration.builder()
+                      .withIdentifier(builder()
+                          .namespace(TLS_PREFIX)
+                          .name(TLS_REVOCATION_CHECK_ELEMENT_IDENTIFIER)
+                          .build());
+
+                  InfrastructureTypeMapping.getQName(objectValue.getTypeId())
```

Comment:
this can be refactored even more, the conditional should cover only this block and the `cloneParameters` in the else. The first and last parts of both if and else blocks are the same. This code is already a mess, so everything that we can simplify is valuable